CMakeLists.txt

```diff
@@ -4,6 +4,27 @@
 
 cmake_minimum_required(VERSION 3.0.2)
 
+#
+# The following policies introduced in CMake 3.15 are needed to work 
+# around a flaw in the Visual Studio CMake integration. The used 
+# toolchain and platform definitions define /W3 and /MD (Release) 
+# respectively /MDd (Debug) by default during the compiler 
+# determination process.
+# If the warning level or the C runtime option are changed afterwards
+# these new options will not replace the already set ones but will be
+# added additionally. This leads to compiler warning D9025 (overriding
+# command line options) which can not be suppressed. These policies 
+# prevent the warning flags and runtime option set by default.
+#
+#  CMP0092 - warnings (https://cmake.org/cmake/help/v3.15/policy/CMP0092.html)
+#  CMP0091 - runtime  (https://cmake.org/cmake/help/v3.15/policy/CMP0091.html)
+#
+if (POLICY CMP0091)
+  cmake_policy(SET CMP0091 NEW)
+endif ()
+if (POLICY CMP0092)
+  cmake_policy(SET CMP0092 NEW)
+endif ()
```

Comment:
There is a flaw in the Visual Studio CMake integration which produces annoying compiler warnings if one tries to set the warning level or the runtime library after the compilation system is detected. This policies tell cmake to not set the options by default.